client-sdk/go/modules/contracts/types.go

```
@@ -179,6 +180,70 @@ type InstanceStorageQueryResult struct {
 	Value []byte `json:"value"`
 }
 
+// StoreKind defines the public or confidential store type for performing queries.
+type StoreKind uint32
+
+// MarshalText returns string representation of StoreKind.
+func (sk StoreKind) MarshalText() string {
+	switch sk {
+	case StoreKindPublic:
+		return "public"
+	case StoreKindConfidential:
+		return "confidential"
+	}
+
+	return ""
```

This should return an error instead.